### PR TITLE
PC-224 Add verification to rest and Go sdk

### PR DESCRIPTION
**Pull request by xgreenx**
Added HashLock and SecretLock routes and models.
Added StartDriveVerificationTransaction and EndDriveVerificationTransactions.
Updated dfms flow integration test.

### CODE REVIEW

File: sdk/lock_model.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+type SecretLockInfo struct {
+	CommonLockInfo
+	HashAlgorithm   HashType
+	CompositeHash   *Hash
+	Secret          *Hash
+	Recipient       *PublicAccount
+}
+
+func CalculateCompositeHash(secret *Hash, recipient *Address) (*Hash, error) {
```
Comment (Wondertan) [inline sdk/lock_model.go]: Can you give an example use case for that func?

File: sdk/storage.go [hunk 2]
```diff
@@ -64,3 +68,42 @@ func (s *StorageService) GetAccountDrives(ctx context.Context, driveKey *PublicA
 
 	return dto.toStruct(s.client.NetworkType())
 }
+
+func (s *StorageService) GetVerificationStatus(ctx context.Context, driveKey *PublicAccount) (*VerificationStatus, error) {
+	if driveKey == nil {
+		return nil, ErrNilAddress
+	}
+
+	compositeHash, err := CalculateCompositeHash(&Hash{}, driveKey.Address)
```
Comment (Wondertan) [inline sdk/storage.go]: Why do we need "composite hash" here?

File: sdk/storage.go [hunk 3]
```diff
@@ -64,3 +68,42 @@ func (s *StorageService) GetAccountDrives(ctx context.Context, driveKey *PublicA
...
+		}
+
+		return nil, err
+	}
+
+	if lockInfo.HashAlgorithm != Internal_Hash_Type {
+		return nil, errors.New("wrong type of drive secret lock")
+	}
+
+	return  &VerificationStatus{
```
Comment (Wondertan) [inline sdk/storage.go]: Remove space)

Comment (xgreenx) [reply sdk/storage.go]: Oh, no! It is terrible change. I will change it at one minute! I must to save the world

**Pull request merged**